### DISCUSSION TITLE
How can I extend the can method?

### DISCUSSION THREAD

**Discussion by anardil** [spatie/laravel-permission#2074] [Q&A]
Hello,

I don't use the team feature but I added on my own a column in the table ..has_permission for account_id.
Now I want to extends the method **can** to add as param the accountId variable.

```
<?php
$can = $user->can('read', $accountId);
?>
```

Do you know how I can do this ?

Thank you.

**Comment by erikn69** [comment]
It is not posible, `can` method comes from laravel framework
if better if you set `permission.register_permission_check_method=false`, and make/register your own `checkPermissionTo` on `Gate`, but still you can't send `$accountId` argument(maybe with a pipe and `explode`)
https://github.com/spatie/laravel-permission/blob/3aab62f59fa23b8f88ff464fe46bf939aa976ae2/src/PermissionRegistrar.php#L131-L140
